go-gitea/gitea
# Unifies pagination template usage (#6531)

This pull request makes the additional URL params that are used in the pagination template configurable. This enables the usage of the same template among all pages that use a pagination.

Resolves #6531.

---

### How to test
Check the paginations on all pages that use one, i.e. templates with `{{template "base/paginate" .}}`

- General
  - Explore repositories: http://localhost:3000/explore/repos
  - Explore users: http://localhost:3000/explore/users
  - Explore organisations: http://localhost:3000/explore/organizations
  - Explore code (redirects to explore/repos?!): http://localhost:3000/explore/code
- User
  - User home: http://localhost:3000
  - User dashboard Issues/Pulls: http://localhost:3000/issues / http://localhost:3000/pulls
  - User profile: http://localhost:3000/[username]
  - User notifications: http://localhost:3000/notifications
  - Organisation profile: http://localhost:3000/[orgname]
- Repository
  - Repository search: http://localhost:3000/[owner]/[repo]/search
  - Issue List: http://localhost:3000/[owner]/[repo]/issues
  - Milestones: http://localhost:3000/[owner]/[repo]/milestones
  - Releases: http://localhost:3000/[owner]/[repo]/releases
  - Milestone details: http://localhost:3000/[owner]/[repo]/milestone/[milestone]
  - template repo/commits_table, e.g. repository commits: http://localhost:3000/[owner]/[repo]/commits/branch/master
- Admin
  - Admin: organisation list: http://localhost:3000/admin/orgs
  - Admin: user list: http://localhost:3000/admin/users
  - Admin: repository list: http://localhost:3000/admin/repos
